### DISCUSSION TITLE
Feature request: better typing windows

### DISCUSSION THREAD

**Discussion by Liampie** [OpenRCT2/OpenRCT2#15712] [Feature requests]
In RCT and RCT2, if you want to type in a ride name, park name, savegame name or a banner text, you would do it in a default windows window, and upon opening this window the current name would be selected.

In OpenRCT you have a wonderful looking custom window, but instead of having the whole text selected, the cursor is at the end of the text. Renaming a ride means you have to backspace the whole text. The cursor is no longer a line inbetween letters either, it's now a letter being underlined which is kinda confusing if you want to add or remove a symbol from the middle of the text. I never know if an underlined symbol means that whatever you do comes before or after the underlined symbol. All in all it's just not a very functional window and it annoys me every time. The functionality the old windows window had is crucial in my experience!

**Comment by Gymnasiast** [comment]
OpenTTD has a button to clear text in such windows.

**Comment by ipatix** [comment]
You can also Ctrl+Backspace to delete all the text. That's good enough for me personally.

**Comment by pizza2004** [comment]
I don't want to be a bother or seem like I'm showing up to complain or anything.  I think that having the text selected so you could just start typing right off is nice, and the kind of text cursor you mention is definitely more usable, but it might add needless complexity to the codebase, so I wouldn't get your hopes up for this one.

**Comment by Gymnasiast** [comment]
Hm, an I-beam cursor is something that improves usability, so I think it's warranted.
